### DISCUSSION TITLE
New Ideas for feautures

### DISCUSSION THREAD

**Discussion by micha003** [trevorbayless/cli-chess#26] [Ideas]
Hi, I already submitted an Issue, but I want to contribute much more to this nice project. So I went through the Issues, but I could not find a fitting Issue for me to work on. @trevorbayless, have you any concrete Plans in the future, which I can probably work on?

**Comment by trevorbayless** [comment]
Thank you very much for your willingness to contribute! I actually don't have any sort of concrete plans with for project at the moment, but am open to any suggestions and will definitely give any PR that comes through a good consideration. If I were to rate the current enhancements by want level, I'd say getting cli-chess integrated with OAuth API followed by having actual ticking clocks would be the top contenders, but not the easiest to start with. Are you new to working with Python?

**Reply by micha003** [reply]
I'm a advanced python programmer, but I never really worked with CLI Apps, can you recommend any tutorials, what do you use exactly?

**Reply by trevorbayless** [reply]
cli-chess uses [prompt-toolkit](https://github.com/prompt-toolkit/python-prompt-toolkit) for the UI. It has very nice [documentation](https://python-prompt-toolkit.readthedocs.io/en/master/) and some great [examples](https://github.com/prompt-toolkit/python-prompt-toolkit/tree/master/examples) as well. The examples and documentation are what I used to get a better understanding of how prompt-toolkit works before I started building this application with it. If you look through the cli-chess source code, you will see models, views, and presenters. The views are where prompt-toolkit is specifically used in this project.

**Comment by zlatistiv** [comment]
Hello,

I've been trying to build my own cli chess as well but isn't nearly as useful as yours.
The thing I'd say I've done well is the move selection, instead of using chess notation you can navigate on the board with a cursor.
If I could I'd submit a PR but I'm not sure how to navigate your code base.

If you could give some guidance and think it's a good idea I'd be glad to contribute.

**Reply by trevorbayless** [reply]
Very cool to hear you're working on a CLI chess application, I'll have to check it out!

I'll gladly review a PR that comes through with click to move functionality, sounds neat. It definitely would be a larger change, and the board view / presenter would need to be modified. Specifically, you'd have to look into how to handle character selection using prompt toolkit (which is definitely possible). As far as how to use prompt toolkit, or the general architecture of the cli-chess code please see this reply: https://github.com/trevorbayless/cli-chess/discussions/26#discussioncomment-7542478
